source/services/dataContracts/dataContractsHelper/dataContractsHelper.service.ts

```diff
@@ -6,18 +6,16 @@ export interface IDataContractsHelper {
 
 class DataContractsHelper implements IDataContractsHelper {
 	versionEndpoint(endpoint: string, versionNumber: number): string {
-		let endpointFragments: string[];
 		let versionExpression: RegExp = /\/v\d+\//;
 
+		let versionString: string = 'v' + versionNumber;
+
 		let searchResult = endpoint.search(versionExpression);
 		if (searchResult !== -1) {
-			endpointFragments = endpoint.split(versionExpression);
+			return endpoint.replace(versionExpression, '/' + versionString + '/');
 		} else {
-			endpointFragments = endpoint.split('api');
-			endpointFragments[0] += 'api/';
+			return endpoint.replace('api', 'api/' + versionString);
```

Comment:
Hate to be a stickler, but we probably want to change this to:
`return endpoint.replace(/^api\//, 'api/ + versionString + '/');`

Otherwise this would also match against something like `'api/test/hapiness'`, changing it into `'api/v2/test/hv2ness'`